Dependancy-Injection/di.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+// Greet sends a greeting to a writer.
+func Greet(writer io.Writer, name string) {
+	// fmt.Fprintf to format the greeting message and write it to the provided writer.
+	//fmt.Fprintf(writer, "Hello, %s", name)
```

Comment:
nit: dead code should be removed instead of commented out - we'll have git history if we want to revisit later.